### DISCUSSION TITLE
unityContext.on(8.8) → ?? (9.0.3)

### DISCUSSION THREAD

**Discussion by devilofdev** [jeffreylanters/react-unity-webgl#404] [Help Wanted!] [answered]
i used below code (react-unity-webgl@8.8.0)
```ts
useEffect(() => {
    unityContext.on("unityToFront", PassOnSessionInfo);
    return function () {
      unityContext.removeAllEventListeners();
      unityContext.quitUnityInstance();
    };
  }, []);
```

in 9.0.3 version, i thought that unityContext.on === addEventListener.. but not working
help!
```ts

useEffect(() => {
    addEventListener("unityToFront", PassOnSessionInfo);
    return function () {
      removeEventListener("unityToFront", PassOnSessionInfo);
    };
  }, []);
```

**Comment by jeffreylanters** [comment, marked as answer]
Hi, don't forget to add both the add and remove event listener as well as you own callback to the dependency array. Let me know if this worked.

**Reply by devilofdev** [reply]
thanks for reply 🥺

i changed the code after reply to me
```ts
useEffect(() => {
    addEventListener("unityToFront", PassOnSessionInfo);
    return function () {
      removeEventListener("unityToFront", PassOnSessionInfo);
    };
  }, [addEventListener, removeEventListener, PassOnSessionInfo])
```

but still error.
<img width="447" alt="Screen Shot 2022-06-29 at 15 53 19" src="https://user-images.githubusercontent.com/46062585/176371457-c8b65a45-eb88-4869-9d83-c96f7ad5c09e.png">

"unityToFront" is the name that Unity calls Javascript function
this code works well in version 8.8.0 like "unityContext.on("unityToFront", PassOnSessionInfo);"

**Reply by jeffreylanters** [reply]
No worries! The `ReactUnityWebGL` object is no longer available via the JSLib in your Unity Project and is replaced with the `dispatchReactUnityEvent` function instead. You have to make some small adjustments to the JSLib in your Unity project. (Don't forget to make a new Unity build afterwards).

The changes should look something like this;

```diff
mergeInto(LibraryManager.library, {
  MyCustomEvent: function (someValue) {
--  ReactUnityWebGL.MyCustomEvent(someValue);
++  window.dispatchReactUnityEvent("MyCustomEvent", someValue);
  },
});
```

For more information on these functions, head over to the [documentation](https://react-unity-webgl.dev/docs/api/event-system/). I hope this helps!

edit: Just like @helloingob suggested.

**Reply by devilofdev** [reply]
thanks 🙆🏻 
it works well.

**Comment by helloingob** [comment]
You have to change the Unity Javascript PLUGIN part aswell.

https://react-unity-webgl.dev/docs/api/event-system

	mergeInto(LibraryManager.library, {
		SetScore: function(score) {
			try {
				window.dispatchReactUnityEvent("unityToFront", PassOnSessionInfo);
			} catch (e) {
				console.warn("Failed to dispatch event");
			}
		},
	});
